File: app/src/main/java/com/dreams/androidquizapp/models/Question.java
package com.dreams.androidquizapp.models;

/**
 * Created by Tc2r on 5/19/2017.
 * <p>
 * Description:
 */

public class Question {
	private final int id;
	private final String question;
	private final String details;
	private final String questionType;
	private final String shortAns;
	private final String trueOrFalse;

	public int getId() {
		return id;
	}

	public String getQuestionType() {
		return questionType;
	}

	public String getQuestion() {
		return question;
	}

	public String getDetails() {
		return details;
	}

	public String getShortAns() {
		return shortAns;
	}

	public String getTrueOrFalse() {
		return trueOrFalse;
	}


	public static class Builder {
		// Required Parameters
		private final int id;
		private final String questionType;
		private final String question;
		private final String details;

		// Optional Parameters - Initialized to default variables.
		private String shortAns = "";
		private String trueOrFalse = "false";

		public Builder(int id, String questionType, String question, String details) {
			this.id = id;
			this.questionType = questionType;
			this.question = question;
			this.details = details;

		}

		public Builder shortAns(String val) {
			shortAns = val;
			return this;
		}

		public Builder trueOrFalse(String val) {
			trueOrFalse = val;
			return this;
		}

		public Question build() {
			return new Question(this);
		}
	}

	private Question(Builder builder) {
		id = builder.id;
		questionType = builder.questionType;
		question = builder.question;
		details = builder.details;
		shortAns = builder.shortAns;
		trueOrFalse = builder.trueOrFalse;
	}
}

File: Android_Interview_Prep_App/app/src/main/java/com/dreams/androidquizapp/models/Question.kt
package com.dreams.androidquizapp.models

/**
 * Created by Tc2r on 5/19/2017.
 *
 *
 * Description:
 */
class Question constructor(builder: Builder) {
    val id: Int
    val question: String
    val details: String
    val questionType: String
    val shortAns: String
    val trueOrFalse: String

    class Builder( // Required Parameters
        val id: Int,
        val questionType: String,
        val question: String,
        val details: String
    ) {
        // Optional Parameters - Initialized to default variables.
        var shortAns = ""
        var trueOrFalse = "false"
        fun shortAns(short: String): Builder {
            shortAns = short
            return this
        }

        fun trueOrFalse(tF: String): Builder {
            trueOrFalse = tF
            return this
        }

        fun build(): Question {
            return Question(this)
        }
    }

    init {
        id = builder.id
        questionType = builder.questionType
        question = builder.question
        details = builder.details
        shortAns = builder.shortAns
        trueOrFalse = builder.trueOrFalse
    }
}
